File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.cpp
class Solution {
public:
    int maxOperations(string s) {
        int operationsCount = 0, onesCount = 0;
        bool hasOne = false;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '1') {
                onesCount++;
                hasOne=1;
            }
            if (s[i] == '0' && hasOne == true) {
                 operationsCount += onesCount;
                 hasOne=0;
            }
        }

        

        return operationsCount;
    }
};

File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        lis=[]
        n=len(s)
        ans=0
        las=-1
        ct=0
        for i in range(n):
            if s[i]=="1":
                if las+1<i :
                    ans+=ct
                las=i
                ct+=1
        if s[-1]!="1":
            ans+=ct        
        return ans            